forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt

```diff
@@ -127,7 +153,48 @@ class ForageTerminalSDK(
     override suspend fun checkBalance(
         params: CheckBalanceParams
     ): ForageApiResponse<String> {
-        return ForageApiResponse.Success("TODO")
+        val logger = createLogger()
+        val (foragePinEditText, paymentMethodRef) = params
+        val (merchantId, sessionToken) = forageSdk._getForageConfigOrThrow(foragePinEditText)
+
+        // TODO: replace Log.getInstance() with Log() in future PR
```

Comment:
Did you actually solve this TODO in this PR through the `createLogger()` function?